### PR TITLE
Refactor preview of osd battery usage

### PR DESCRIPTION
**Pull request by McGiverGim**
As discussed with @chmelevskij and @haslinghuis here: https://github.com/betaflight/betaflight-configurator/pull/2468 this PR does a simply refactor of the preview for osd battery usage.

I liked the @chmelevskij idea, but after implementing @haslinghuis one, I think the first is not necessary. It needs to declare some vars that will be used only at some cases of the switch. After using the `repeat` I think is enough.

### CODE REVIEW

**Review by haslinghuis — APPROVED** [review]

**Review by chmelevskij — APPROVED** [review]

**Review by Asizon — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
